modules/nf-core/delly/filter/main.nf

```diff
@@ -0,0 +1,71 @@
+process DELLY_FILTER {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/delly:1.2.6--hb7e2ac5_0':
+        'biocontainers/delly:1.2.6--hb7e2ac5_0' }"
+
+    input:
+    tuple val(meta), path(bcf), path(csi)
+    val(mode)
+    path(samples)
+
+    output:
+    tuple val(meta), path("${prefix}.bcf")    , emit: bcf
+    tuple val(meta), path("${prefix}.bcf.csi"), emit: csi
+    path "versions.yml"                       , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    if ("$bcf" == "${prefix}.bcf") error "Input and output names are the same, set prefix in module configuration to disambiguate!"
+    def samples_arg = ''
+    def sample_list = ''
+    if (mode == 'somatic') {
+        if (!samples && meta.control_samples && meta.tumor_samples) {
+            sample_list = meta.control_samples.collect {"${it}\tcontrol"}.join('\n')
+            sample_list += '\n' + meta.tumor_samples.collect {"${it}\ttumor"}.join('\n')
+        }
```

Comment:
`meta.control_samples` and `meta.tumor_samples` aren't required fields in a meta map and thus can't be used in this module. Maybe you can solve this by splitting the `samples` channel into `tumor_samples` and `normal_samples`

Reply:
Providing a sample description file in `samples` alone is sufficient to run the module. I mainly added the option to also be able to specificy sample labels in the meta map to avoid having to provide a sample description file for the tests. Otherwise, I would need to upload one to test-datasets. Or do you think it can also be included in the tests folder?

Reply:
You can just open a simple PR to the test datasets repository, it doesn't take long :). Please ping me and I'll approve it

Reply:
What do you think is better from the point of view of the user as input for the module?
1) `path(file_with_sample_labels)` --> The user needs to provide a properly formatted TSV file with sample labels.
2) `val(tumor_samples)` and `val(normal_samples)` --> The user just needs to provide lists of sample ids and the TSV file is generated in the module.

Reply:
Maybe both?

Reply:
Users can now either provide a sample description file or provide sample ids for tumor and control samples as lists.